File: 프로그래머스/prev/Level1&2/최소직사각형.py
def solution(sizes):
    for i in range(len(sizes)):
        if sizes[i][0] < sizes[i][1]:
            sizes[i][0], sizes[i][1] = sizes[i][1], sizes[i][0]

    return sorted(list(zip(*sizes))[0])[-1] * sorted(list(zip(*sizes))[1])[-1]

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
import java.util.*;

class Solution {
    
    public int solution(int[][] sizes) {
        int maxWidth = 0;
        int maxHeight = 0;

        for (int[] size : sizes) {
            int w = Math.max(size[0], size[1]);
            int h = Math.min(size[0], size[1]);

            if (w > maxWidth) maxWidth = w;
            if (h > maxHeight) maxHeight = h;
        }

        return maxWidth * maxHeight;
    }
}